docs/design/coreclr/jit/hot-cold-splitting.md

```diff
@@ -0,0 +1,173 @@
+# Hot/Cold Code Splitting
+
+This document describes the current state of hot/cold splitting in the JIT.
+
+Hot/Cold splitting is an optimization where generated code is split into frequently-executed ("hot") and
+rarely-executed ("cold") parts and placed in separate memory regions. Increased hot code density better leverages
```

Comment:
Nit -- keep an eye out for passive voice and try to avoid it where you can. Here it seems pretty easy:
```suggestion
Hot/Cold splitting is an optimization that splits code into frequently-executed ("hot") and
rarely-executed ("cold") parts and places the parts in separate memory regions. Increased hot code density better leverages
```